engine/src/flutter/lib/web_ui/lib/src/engine/semantics/label_and_value.dart

```diff
@@ -491,12 +501,96 @@ class LabelAndValue extends SemanticBehavior {
     return computeDomSemanticsLabel(
       tooltip: semanticsObject.hasTooltip ? semanticsObject.tooltip : null,
       label: semanticsObject.hasLabel ? semanticsObject.label : null,
-      hint: semanticsObject.hint,
       value: shouldDisplayValue ? semanticsObject.value : null,
     );
   }
 
+  void _updateHintDescription(String? hint) {
+    if (hint != null && hint.trim().isNotEmpty) {
+      _setAriaDescriptionOrDescribedBy(hint);
+    } else {
+      owner.removeAttribute('aria-description');
+      owner.removeAttribute('aria-describedby');
+    }
+  }
+
+  void _setAriaDescriptionOrDescribedBy(String hint) {
+    if (supportsAriaDescription) {
+      owner.setAttribute('aria-description', hint);
+      owner.removeAttribute('aria-describedby');
+    } else {
+      final String id = _ensureDescribedBy(hint);
+      owner.setAttribute('aria-describedby', id);
+      owner.removeAttribute('aria-description');
+    }
+  }
+
+  String _ensureDescribedBy(String hint) {
+    _describedBySpan ??= domDocument.createElement('span');
+    _describedBySpan!.id = 'flt-hint-${semanticsObject.id}';
+    _describedBySpan!.setAttribute('hidden', '');
+    _describedBySpan!.text = hint;
+    if (!(_describedBySpan?.isConnected ?? false)) {
+      // Append the describedby span as a sibling to the semantics element,
+      // not as a child. This is crucial because if the span were a child,
+      // element.text would read both the label and hint text concatenated
+      // together (e.g., "LabelHint" instead of just "Label").
+      //
+      // DOM structure we want:
+      //   <parent>
+      //     <flt-semantics>Label</flt-semantics>
+      //     <span hidden aria-hidden="true">Hint</span>
+      //   </parent>
+      //
+      // This ensures element.text on the semantics element returns only "Label"
+      // while aria-describedby still works correctly since it references by ID.
+      final DomElement? parent = owner.element.parentElement;
+      if (parent != null) {
+        parent.append(_describedBySpan!);
+      } else {
+        // Fallback: append to document.body if no parent is available.
+        // This can happen during DOM construction or if the element is detached.
+        // aria-describedby will still work since it references the span by ID.
+        domDocument.body!.append(_describedBySpan!);
+      }
+    }
+    return _describedById!;
+  }
+
+  static bool? _supportsAriaDescription;
+
+  static bool get supportsAriaDescription {
+    _supportsAriaDescription ??= _checkAriaDescriptionSupport();
+    return _supportsAriaDescription!;
+  }
+
+  @visibleForTesting
+  static set supportsAriaDescriptionForTest(bool? value) {
+    _supportsAriaDescription = value;
+  }
+
+  static bool _checkAriaDescriptionSupport() {
+    if (isChromium || isEdge) {
+      return true;
+    }
+    if (isSafari174OrNewer) {
+      return true;
+    }
+    if (isFirefox119OrNewer) {
+      return true;
+    }
+    return false;
```

Comment:
Now that all are booleans, you can be more concise 🙂
```suggestion
    return isChromium || isEdge || isSafari174OrNewer || isFirefox119OrNewer;
```